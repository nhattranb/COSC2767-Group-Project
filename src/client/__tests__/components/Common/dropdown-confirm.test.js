import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import DropdownConfirm from '../../../app/components/Common/DropdownConfirm';
import { UncontrolledButtonDropdown, DropdownToggle, DropdownMenu } from 'reactstrap';
import {describe, expect, it, jest} from "@jest/globals";
import '@testing-library/jest-dom';

jest.mock('reactstrap', () => ({
    UncontrolledButtonDropdown: ({ children }) => <div>{children}</div>,
    DropdownToggle: ({ children, ...props }) => <button {...props}>{children}</button>,
    DropdownMenu: ({ children }) => <div>{children}</div>,
}));

describe('DropdownConfirm component', () => {
    it('renders correctly with default props', () => {
        const { asFragment } = render(<DropdownConfirm />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('renders correctly with custom className', () => {
        const { asFragment } = render(<DropdownConfirm className="custom-class" />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('renders correctly with label', () => {
        const { asFragment } = render(<DropdownConfirm label="Test Label" />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('renders correctly with children', () => {
        const { asFragment } = render(
            <DropdownConfirm>
                <p>Additional content</p>
            </DropdownConfirm>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
