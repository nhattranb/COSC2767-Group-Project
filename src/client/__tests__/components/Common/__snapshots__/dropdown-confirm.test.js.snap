// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownConfirm component renders correctly with children 1`] = `
<DocumentFragment>
  <div
    class="dropdown-confirm undefined"
  >
    <div>
      <button>
        <div
          class="dropdown-action sm"
        >
          <span
            class="fa fa-chevron-down dropdown-caret"
          />
        </div>
      </button>
      <div>
        <p>
          Additional content
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`DropdownConfirm component renders correctly with custom className 1`] = `
<DocumentFragment>
  <div
    class="dropdown-confirm custom-class"
  >
    <div>
      <button>
        <div
          class="dropdown-action sm"
        >
          <span
            class="fa fa-chevron-down dropdown-caret"
          />
        </div>
      </button>
      <div />
    </div>
  </div>
</DocumentFragment>
`;

exports[`DropdownConfirm component renders correctly with default props 1`] = `
<DocumentFragment>
  <div
    class="dropdown-confirm undefined"
  >
    <div>
      <button>
        <div
          class="dropdown-action sm"
        >
          <span
            class="fa fa-chevron-down dropdown-caret"
          />
        </div>
      </button>
      <div />
    </div>
  </div>
</DocumentFragment>
`;

exports[`DropdownConfirm component renders correctly with label 1`] = `
<DocumentFragment>
  <div
    class="dropdown-confirm undefined"
  >
    <div>
      <button>
        <div
          class="dropdown-action sm"
        >
          Test Label
          <span
            class="fa fa-chevron-down dropdown-caret"
          />
        </div>
      </button>
      <div />
    </div>
  </div>
</DocumentFragment>
`;
